core/src/builtin_widgets/padding.rs

```diff
@@ -33,9 +31,33 @@ impl Declare for Padding {
   fn declarer() -> Self::Builder { FatObj::new(()) }
 }
 
-impl Render for Padding {
-  fn perform_layout(&self, clamp: BoxClamp, ctx: &mut LayoutCtx) -> Size {
-    space_around_layout(&self.padding, &clamp, ctx)
+impl_compose_child_for_wrap_render!(Padding, DirtyPhase::Layout);
+
+impl WrapRender for Padding {
+  fn perform_layout(&self, clamp: BoxClamp, host: &dyn Render, ctx: &mut LayoutCtx) -> Size {
+    let thickness = self.padding.thickness();
+
+    let min = (clamp.min - thickness).max(ZERO_SIZE);
+    let max = (clamp.max - thickness).max(ZERO_SIZE);
+    // Shrink the clamp of child.
+    let child_clamp = BoxClamp { min, max };
+    let size = host.perform_layout(child_clamp, ctx);
+    clamp.clamp(size + thickness)
+  }
+
+  fn paint(&self, host: &dyn Render, ctx: &mut PaintingCtx) {
+    ctx.content_only_transform_apply(&Transform::translation(self.padding.left, self.padding.top));
+    host.paint(ctx);
+  }
+
+  fn get_transform(&self, host: &dyn Render) -> Option<Transform> {
```

Comment:
The transform now is used to map to parent, it seems have some bug to get_transform with wrap render which adjust some part of the host but not the entire. Here the padding wrap render will offset the host by the top, left padding, but if the position is in the padding space, it will also be offset by the top left padding, which will be map to Inaccurate position

Reply:
When transform is used to map a position, it is relative to the parent or child. In this scenario, there is no need to calculate a position for a part of the widget.